usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -192,6 +195,50 @@ spa_history_zone(void)
 }
 
 /*
+ * Post a history sysevent.  The nvlist_t* passed into this function will be
+ * duplicated (by nvlist_add_nvlist) and should be freed by caller.
+ * This function will only run in the context of the kernel.
+ */
+static void
+spa_history_log_notify(spa_t *spa, nvlist_t *nvl)
```

Comment:
I've pushed a new commit (this can be squashed at the end, but for now it makes it easier to see the intermediate diff) that

1. modifies `spa_event_notify` and `spa_event_create` to take a `hist_nvl` argument, but doesn't modify whether the existing function was static or not (as I only need to use the `spa_event_notify` wrapper)
2. changes the argument name to `hist_nvl` to show what its purpose is